### PR TITLE
Prevent guests to add a downloadable product #9542

### PR DESCRIPTION
**Pull request by AmooAti**
## Issue Reference
#9542 

## Description
Hi, I've implemented the requested enhancement and removed some tests regarding testing downloadable products when the customer is not logged in.
But maybe having a `redirect_url` in the cookie to redirect customers back to the product page has a better UX.

## How To Test This?
1. Create a downloadable product.
2. When you are a guest, try to add an item to the cart using the store front.

### CODE REVIEW

File: packages/Webkul/Shop/tests/Feature/Checkout/CartTest.php [hunk 1]
```diff
@@ -576,6 +576,8 @@
 
 it('should add a downloadable product to the cart', function () {
     // Arrange
+    $this->loginAsCustomer();
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/tests/Feature/Checkout/CartTest.php:579]: This is part of act not arrange. This line shoule be after act and assert comment.

File: packages/Webkul/Shop/tests/Feature/Product/Prices/DownloadableTest.php [hunk 2]
```diff
@@ -83,6 +83,8 @@
 
 it('should add a downloadable product to the cart with a cart rule of the no coupon type for all customer group type', function () {
     // Arrange
+    $this->loginAsCustomer();
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/tests/Feature/Product/Prices/DownloadableTest.php:86]: Same here.

File: packages/Webkul/Shop/src/Resources/views/products/view.blade.php [hunk 3]
```diff
@@ -487,10 +487,16 @@ class="icon-compare text-2xl"
                                     this.$emitter.emit('add-flash', { type: 'success', message: response.data.message });
 
                                     if (response.data.redirect) {
-                                        window.location.href= response.data.redirect;
+                                        window.location.href = response.data.redirect;
                                     }
                                 } else {
                                     this.$emitter.emit('add-flash', { type: 'warning', message: response.data.data.message });
+
+                                    if(response.data.data.redirect) {
```
Comment (devansh-webkul) [inline packages/Webkul/Shop/src/Resources/views/products/view.blade.php:495]: Add space after if.

**Review by devansh-webkul — CHANGES_REQUESTED** [review]
Hello @AmooAti,

Thank you for the contribution, please look at the minor changes.

**Comment by AmooAti** [comment]
Hello @devansh-webkul 
Thank you for your review. It was an excellent point, and I appreciate it.
I've just made changes.

**Comment by devansh-webkul** [comment]
Hi @AmooAti,

I just discussed this feature, but as per ou flow we dont need this PR because we are having different flow here. Guest user can add to cart at the time of checkout we use login feature there.

Sorry for this, but this pull request as of now cant be merged.

**Pull request closed without merging**